### ISSUE TITLE
Replication tests fail sporadically with "User 'cluster' is not found"

### ISSUE THREAD

**Issue by locker** [tarantool/tarantool#3637]
Reproduced by [Travis CI](https://travis-ci.org/tarantool/tarantool/jobs/419163281) on 1.10:
```
[011] replication/ddl.test.lua                        memtx           [ pass ]
[011] replication/ddl.test.lua                        vinyl           
[011] 
[011] [Instance "ddl4" returns with non-zero exit code: 1]
[011] 
[011] Last 15 lines of Tarantool Log file [Instance "ddl4"][/tarantool/test/var/011_replication/ddl4.log]:
[011] 2018-08-22 13:32:24.465 [10448] main/101/ddl4 C> log level 5
[011] 2018-08-22 13:32:24.465 [10448] main/101/ddl4 I> mapping 268435456 bytes for memtx tuple arena...
[011] 2018-08-22 13:32:24.465 [10448] main/101/ddl4 I> mapping 134217728 bytes for vinyl tuple arena...
[011] 2018-08-22 13:32:24.468 [10448] iproto/101/main I> binary: bound to unix/:/tarantool/test/var/011_replication/autobootstrap4.sock
[011] 2018-08-22 13:32:24.468 [10448] main/108/applier/cluster@unix/:/tarantoo I> remote master is 1.10.1 at unix/:/tarantool/test/var/011_replication/autobootstrap2.sock
[011] 2018-08-22 13:32:24.470 [10448] main/109/applier/cluster@unix/:/tarantoo I> remote master is 1.10.1 at unix/:/tarantool/test/var/011_replication/autobootstrap3.sock
[011] 2018-08-22 13:32:24.470 [10448] main/107/applier/cluster@unix/:/tarantoo I> remote master is 1.10.1 at unix/:/tarantool/test/var/011_replication/autobootstrap1.sock
[011] 2018-08-22 13:32:24.470 [10448] main/110/applier/cluster@unix/:/tarantoo I> remote master is 1.10.1 at unix/:/tarantool/test/var/011_replication/autobootstrap4.sock
[011] 2018-08-22 13:32:24.470 [10448] main/107/applier/cluster@unix/:/tarantoo I> failed to authenticate
[011] 2018-08-22 13:32:24.470 [10448] main/107/applier/cluster@unix/:/tarantoo xrow.c:891 E> ER_LOADING: Instance bootstrap hasn't finished yet
[011] 2018-08-22 13:32:24.582 [10448] main/107/applier/cluster@unix/:/tarantoo I> failed to authenticate
[011] 2018-08-22 13:32:24.582 [10448] main/107/applier/cluster@unix/:/tarantoo xrow.c:891 E> ER_NO_SUCH_USER: User 'cluster' is not found
[011] 2018-08-22 13:32:24.582 [10448] main/101/ddl4 xrow.c:891 E> ER_NO_SUCH_USER: User 'cluster' is not found
[011] 2018-08-22 13:32:24.582 [10448] main/101/ddl4 F> can't initialize storage: User 'cluster' is not found
[011] 2018-08-22 13:32:24.582 [10448] main/101/ddl4 F> can't initialize storage: User 'cluster' is not found
[011] [ fail ]
[Main process] Got failed test; gently terminate all workers...
[011] Worker "011_replication" got failed test; stopping the server...
```

**Comment by sergw** [comment]
I'm reproducing it using: `TEST_RUN_TESTS=$(for i in $(seq 1 20); do echo -n 'replication/autobootstrap.test.lua '; done) make test`
When changing `TIMEOUT` at `autobootstrap.lua` a little bit more (for example `0.2` instead of `0.1`) then tests passed ok.
But after changing `TIMEOUT` to larger value (example `10`) test slow down dramaticlly.

**Comment by sergepetrenko** [comment]
This can be reproduced on every start with such script:
```
box.cfg({
    listen=3301,
    replication={
        "cluster:pass@localhost:3301",
        "cluster:pass@localhost:3302"
    },  
    replication_timeout=0.1,
    replication_connect_timeout=2.0
})

box.once("bootstrap", function()
    require("fiber").sleep(1)
    box.schema.user.create("cluster", {password="pass"})
    box.schema.user.grant("cluster", "replication")
    box.schema.space.create("test")
    box.space.test:create_index("primary")
end)
```
Second instance has the same script except that `listen=3302` (obviously).
My guess is the issue happens every time when `replication_timeout` is smaller than `box.once()` execution time.

**Comment by locker** [comment]
Fixed by 33950162f3e766d413567cd75aaa7e6c384831bd
